pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -78,6 +79,12 @@ type TemplateHolder interface {
 	IsResolvable() bool
 }
 
+// TemplateStorage is an interface of template storage getter and setter.
+type TemplateStorage interface {
```

Comment:
I'm sure I'm missing something obvious, but could you help me understand why this is implemented in both `Workflow` and `wfOperationCtx`? It seems like all the implementation for `wfOperationCtx` does is call the implementation of the `Workflow`:

```golang
// GetStoredTemplate retrieves a template from stored templates of the workflow.
func (woc *wfOperationCtx) GetStoredTemplate(templateScope string, holder wfv1.TemplateHolder) *wfv1.Template {
	return woc.wf.GetStoredTemplate(templateScope, holder)
}
```

Why do you not pass the `Workflow` as the `TemplateStorage` instead of the `wfOperationCtx` in `operator.go`:

```golang
	woc.tmplCtx = templateresolution.NewContext(wfc.wftmplInformer.Lister().WorkflowTemplates(wf.Namespace), wf, &woc)
```

Like you do in `validate.go`:

```golang
tmplCtx := templateresolution.NewContextFromClientset(wfClientset.ArgoprojV1alpha1().WorkflowTemplates(namespace), wf, wf)
```

Sorry if this is an obvious question.

Reply:
That's because I wanted to let the operator save the workflow.
https://github.com/argoproj/argo/blob/fae7382686d917d78e3909d1f6db79c272a1aa11/workflow/controller/operator.go#L381